controllers/flowlogspipeline/flp_objects.go

```diff
@@ -252,13 +265,83 @@ func (b *builder) addTransformStages(lastStage *config.PipelineBuilderStage) {
 	}
 
 	// prometheus stage (encode) configuration
+	// the aggregate and prometheus rules will be incorporated from the metrics definitions file
 	agg := enrichedStage.Aggregate("aggregate", []api.AggregateDefinition{})
 	agg.EncodePrometheus("prometheus", api.PromEncode{
 		Port:   int(b.desired.PrometheusPort),
-		Prefix: "flp_",
+		Prefix: "netobserv_",
 	})
 }
 
+var generateStages = []string{"extract_aggregate", "encode_prom"}
+
+func (b *builder) mergeMetricsConfiguration(pipeline config.PipelineBuilderStage) {
+	// set confgen.Opt.SrcFolder, etc
+	confgen.Opt.SrcFolder = b.desired.MetricsDefinitionsDir
+	confgen.Opt.DestConfFile = "/dev/null"
+	confgen.Opt.SkipWithTags = b.desired.IgnoreMetrics
+	confgen.Opt.GenerateStages = generateStages
+
+	if confgen.Opt.SrcFolder == "" {
+		return
+	}
+	// run the confgenerator to produce the proper flp configuration for metrics from metrics_definitions
+	cg, err := confgen.NewConfGen()
+	if err != nil {
+		log.Fatalf("failed to initialize NewConfGen %s", err)
+		return
+	}
+	err = cg.Run()
+	if err != nil {
+		log.Fatalf("failed to initialize NewConfGen %s", err)
+		return
+	}
+	truncatedConfig := cg.GenerateTruncatedConfig(generateStages)
+	yFileBytes, err := yaml.Marshal(truncatedConfig)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+	var metricsConfig struct {
+		Parameters []config.StageParam
+	}
+	err2 := yaml.Unmarshal(yFileBytes, &metricsConfig)
+	if err2 != nil {
+		log.Fatal(err2)
+		return
+	}
```

Comment:
Why do we need marshal / unmarshal the config? Couldn't we just use the initial `truncatedConfig` object ?

Reply:
I see that confgen returns a generic map .. could we have it return a typed model instead?

Reply:
changed the confgenerator to return typed model and integrated directly without Marshal/UnMarshal.